design/dsl/endpoint_test.go

```diff
@@ -151,7 +151,7 @@ func assertEndpointMetaData(t *testing.T, actual design.MetadataExpr, expected d
 
 func assertEndpointRequestResponse(t *testing.T, assertType string, actual design.DataType, expected *design.UserTypeExpr) {
 	ut, ok := actual.(*design.UserTypeExpr)
-	if !ok {
+	if !ok || ut == nil {
```

Comment:
`Errorf` doesn't stop the function so you also need to test for `nil` below otherwise the function will `panic`.